corehq/apps/reports/util.py

```diff
@@ -583,25 +583,29 @@ def _sync_tableau_users_with_hq(session, domain):
                 _delete_user_local(session, tableau_user_name)
 
     def _sync_tableau_users_with_remote(session):
+        # Setup
+        def _get_HQ_group_users(session):
+            remote_HQ_group_id = _get_hq_group_id(session)
+            if remote_HQ_group_id:
+                remote_HQ_group_users = session.get_users_in_group(remote_HQ_group_id)
+            else:
+                session.create_group(HQ_TABLEAU_GROUP_NAME, "Viewer")
+                remote_HQ_group_users = []
+            return remote_HQ_group_users
 
-        # Setup - parse/get remote group ID and users in group
-        remote_HQ_group_id = _get_hq_group_id(session)
-        if remote_HQ_group_id:
-            remote_HQ_group_users = session.get_users_in_group(remote_HQ_group_id)
-        else:
-            remote_HQ_group_id = session.create_group(HQ_TABLEAU_GROUP_NAME, "Viewer")
-            remote_HQ_group_users = []
+        all_remote_users = session.get_users_on_site()
```

Comment:
any sense of the scale of this call?  Is there a risk of this being a large enough list to overwhelm the system in some way?

Reply:
"A couple hundred" should be fine, I'd guess.  I'm always cautious when dealing with "all forms/cases/users" in a domain at once, but I think web users are typically an order of magnitude or two less numerous than mobile workers.  No idea what a hard limit looks like, but some places I'd think about it off the top of my head are:

* Showing all of them to the user (probably want pagination)
* Embedding all of them in a querystring (we'd hit the 4k character limit FAST)
* Sending to a DB query (do they go in the request body or the querystring?  How does the DB handle this?)
* Receiving all of them at once from a DB (does the DB call return only a field or two or the entire object?)
* Saving them all at once
* Putting them in a single spreadsheet (the XLS limit is like 64k, XLSX is around a million, IIRC)
* Holding them all in memory (depends on whether it's just IDs or whole objects, tolerance here is pretty high, but we have ran into issues, even with just IDs, for example, that's where [this logic](https://github.com/dimagi/commcare-hq/blob/e9f626810de7b6217dcfd7ff07d5be877c294190/corehq/elastic.py#L77-L87) came from)

In this specific case, if the fetch from Tableau goes through, everything else should be fine.